fix: Handle addressPrefixes during subnet import (Issue #48)

haflidif opened the pull request:
## Description

This PR fixes the subnet import issue where `terraform import` would fail with "Unsupported attribute: addressPrefixes" error.

**Root Cause:** During import operations, `azapi_resource.subnet[0].output.properties` only contains `provisioningState` and doesn't include address prefix information. The module's outputs.tf was only checking `output.properties`, causing the error.

**Solution:** Added a three-tier fallback logic in `modules/subnet/outputs.tf`:
1. Try `output.properties.addressPrefixes` (normal operation)
2. Try `output.properties.addressPrefix` (single prefix format)
3. Fallback to `body.properties.addressPrefixes` (import scenario)
4. Final fallback to `body.properties.addressPrefix`

This ensures address prefixes are accessible during import while maintaining full backwards compatibility with existing deployments.

**Testing:** Verified both import scenarios and normal subnet creation work without regressions.

Fixes #48

## Type of Change

- [ ] Non-module change (e.g. CI/CD, documentation, etc.)
- [x] Azure Verified Module updates:
  - [x] Bugfix containing backwards compatible bug fixes
    - [x] Someone has opened a bug report issue, and I have included "Closes #48" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates.
  - [ ] Breaking changes.
  - [ ] Update to documentation

# Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] I did run all pre-commit checks

<!--  Please keep up to date with the contribution guide at https://aka.ms/avm/contribute/terraform -->

haflidif committed:
fix: Handle addressPrefixes during subnet import (Issue #48)

- Added fallback to body.properties when output.properties is incomplete
- Handles both addressPrefixes (array) and addressPrefix (string) formats
- Fixes 'Unsupported attribute' error during subnet import operations
- Import operations now correctly read address prefixes from state

The azapi_resource.subnet output.properties only contains provisioningState
during import operations. This fix adds a three-tier fallback logic to read
from body.properties when output.properties doesn't contain address information.

Fixes #48

jaredfholgate reviewed (commented):
A potential improvement.

haflidif committed:
fix: Simplify address_prefixes output with multi-argument try()

Replace nested ternary operators with cleaner multi-argument try() function for retrieving subnet address prefixes. This maintains the same fallback logic (output.properties -> body.properties, addressPrefixes -> addressPrefix) but with improved readability.

Fixes #48

haflidif committed:
chore: Update avm script from pre-commit

haflidif commented on modules/subnet/outputs.tf:
```diff
@@ -1,6 +1,16 @@
-  value       = local.ipam_enabled ? azapi_resource.subnet_ipam[0].output.properties.addressPrefixes : try(azapi_resource.subnet[0].output.properties.addressPrefixes, [azapi_resource.subnet[0].output.properties.addressPrefix])
+  value = local.ipam_enabled ? azapi_resource.subnet_ipam[0].output.properties.addressPrefixes : (
+    # Try to get from output.properties first (normal operation)
+    try(azapi_resource.subnet[0].output.properties.addressPrefixes, null) != null ?
```
Sound good @jaredfholgate , I've implemented the fix, tested it and it works like a charm, thanks!

jaredfholgate reviewed (dismissed):
LGTM

haflidif committed:
fix: simplify region selection and add subnet peering prerequisites

- Simplify region_selection.tf files to use is_recommended=true filter
- Remove hardcoded region lists from most examples (reduced from ~60 lines to ~10)
- Keep IPAM-specific region lists for examples requiring AVNM IPAM support
- Add prerequisites.tf for subnet peering examples requiring AllowMultiplePeeringLinksBetweenVnets feature
- Add depends_on to ensure feature registration completes before peering creation

This fixes test failures caused by feature registration requirements and simplifies region management across examples.

haflidif committed:
chore: update avm scripts with config support

- Add support for avm.config.json to load local environment variables
- Add AVM_PORCH_STDOUT environment variable support
- Sync updates from upstream avm tooling

These changes were auto-generated by running pre-commit hooks.

haflidif committed:
fix: example error - add azapi provider requirement for subnet peering example

jaredfholgate reviewed (approved):
LGTM, great job
